### ISSUE TITLE
Pylance not starting if python file is located on UNC network drive.

### ISSUE THREAD

**Issue by VadeLight** [microsoft/pylance-release#4862]
<h3> Environment data </h3>

-   Language Server version: 2023.9.20
-   OS and version: win32 x64
-   Python version (and distribution if applicable, e.g. Anaconda): 
-   python.analysis.indexing: true
-   python.analysis.typeCheckingMode: off

<h3> Code Snippet </h3>

```python
import sys
```

<h3> Repro Steps </h3>

1. Create a new file located on a network share, which is mounted as a network drive in windows explorer "with a drive letter".
2. Add a simple import like the one above.
3. Pylance wont start up and highlight the code or provide code suggestions.

The issue is gone if the language server is downgraded to version 2023.9.10. The Problem is most certainly connected to a new handling of network shares or the resolving of system paths in pylance, as the debug output of the old language server reports paths with drive letters (see logs below), whereas the new pylance version reports UNC network paths (both for the exact same folder opened with visual code).

Note: Disabling security.restrictUNCAccess in Visual Studio Code or adding the UNC host to security.allowedUNCHosts does not fix the issue.

<h3> Expected behavior </h3>

Pylance starts up and highlights the code in some way.

<h3> Actual behavior </h3>

Pylance doesnt even start up.

<h3> Logs </h3>

<h4> Language Server Version 2023.9.20 </h4>

```
2023-09-21 16:24:52.881 [info] [Info  - 4:24:52 PM] (13100) Pylance language server 2023.9.20 (pyright 35eaeca4) starting
2023-09-21 16:24:52.882 [info] [Info  - 4:24:52 PM] (13100) Server root directory: C:\Users\<user>\.vscode\extensions\ms-python.vscode-pylance-2023.9.20\dist
2023-09-21 16:24:52.882 [info] [Info  - 4:24:52 PM] (13100) Starting service instance "Test"
2023-09-21 16:24:52.964 [info] [Info  - 4:24:52 PM] (13100) Setting pythonPath for service "Test": "C:\Program Files\Python310\python.exe"
2023-09-21 16:24:52.964 [info] [Info  - 4:24:52 PM] (13100) Setting pythonPath for service "Test": "C:\Program Files\Python310\python.exe"
2023-09-21 16:24:52.975 [info] [Warn  - 4:24:52 PM] (13100) stubPath \\unchost.de\path\to\dir\Test\typings is not a valid directory.
2023-09-21 16:24:53.144 [info] [Info  - 4:24:53 PM] (13100) Assuming Python version 3.10
2023-09-21 16:24:53.844 [info] [Info  - 4:24:53 PM] (13100) No source files found.
2023-09-21 16:24:59.040 [info] [Info  - 4:24:59 PM] (13100) [IDX(2)] Long operation: scan packages \\unchost.de\path\to\dir\Test\ (4047ms)
```
Note: the paths were slightly adjusted for privacy reasons.

Nothing is logged after this. The 3 files which were located for this test inside the 'Test' directory where not found by pylance.

<h4> Language Server Version 2023.9.10 </h4>

```
2023-09-21 16:43:16.288 [info] [Info  - 4:43:16 PM] (16264) Pylance language server 2023.9.10 (pyright c70adefc) starting
2023-09-21 16:43:16.288 [info] [Info  - 4:43:16 PM] (16264) Server root directory: c:\Users\<user>\.vscode\extensions\ms-python.vscode-pylance-2023.9.10\dist
2023-09-21 16:43:16.289 [info] [Info  - 4:43:16 PM] (16264) Starting service instance "Test"
2023-09-21 16:43:16.363 [info] [Info  - 4:43:16 PM] (16264) Setting pythonPath for service "Test": "C:\Program Files\Python310\python.exe"
2023-09-21 16:43:16.363 [info] [Info  - 4:43:16 PM] (16264) Setting pythonPath for service "Test": "C:\Program Files\Python310\python.exe"
2023-09-21 16:43:16.365 [info] [Warn  - 4:43:16 PM] (16264) stubPath g:\path\to\dir\Test\typings is not a valid directory.
2023-09-21 16:43:16.514 [info] [Info  - 4:43:16 PM] (16264) Assuming Python version 3.10
2023-09-21 16:43:16.712 [info] [Info  - 4:43:16 PM] (16264) Found 3 source files
2023-09-21 16:43:25.452 [info] [Info  - 4:43:25 PM] (16264) [IDX(2)] Long operation: index execution environment g:\path\to\dir\Test (6467ms)
2023-09-21 16:43:25.453 [info] [Info  - 4:43:25 PM] (16264) [IDX(2)] Long operation: index packages g:\path\to\dir\Test (6651ms)
```
Note: the paths were slightly adjusted for privacy reasons.

Afterwards, the code is highlighted correctly.
The 3 files which were located for this test inside the 'Test' directory where found by pylance.

**Comment by rchiodo** [comment]
Does upgrading to the latest prerelease resolve the issue? 2023.9.21? 

There was a problem with symbolic links in 2023.9.20 that may be related to this UNC issue.

**Comment by debonte** [comment]
It still repros for me on 2023.9.21.

**Comment by nathanm** [comment]
Still a bug in 2023.9.21. I also tried the pre-release versions between the official releases, 2023.9.11 and 2023.9.12, which both have the same bug, at least regarding syntax highlighting.

**Comment by peak-cbackus** [comment]
I'm running into the same issue on 2023.9.20 and 2023.9.21. I don't believe the issue is it not starting, as my CPU usage spikes and fan starts spinning like crazy. It stops and CPU usage returns to normal as soon as I close vscode, or switch to 2023.9.10. Any .py file I open gives me an "analyzing" notification and it never finishes. I let it go for 3 hours today to see if it resolved itself, no luck. It was definitely doing something in the logs but unfortunately I accidently deleted those un-installing and re-installing pylance. It seemed to be taking a long time to get through references to large packages like matplotlib and pandas from what I remember.

My initial log looks the same as above for both versions. Doesn't find any source files in 2023.9.20 or 2023.9.21, and finds 81 source files in 2023.9.10.

I'm also running windows 10 64-bit, python 3.8.10 from the Microsoft Store. All packages installed through pip.

I'll also be using 2023.9.10 for the time being.

**Comment by debonte** [comment]
This is a problem with the filter we're using to determine if a file should be included in analysis. Pylance is starting, but it fails to find any user files. The problem is specific to UNC paths. It's not clear to me that we've ever supported UNC paths properly. A recent change that we made resulted in network drives (ex. `z:\Test`) being translated into their UNC equivalent (ex. `\\machine\share\Test`) and this is causing users to hit the UNC issue.

**Comment by peak-cbackus** [comment]
> This is a problem with the filter we're using to determine if a file should be included in analysis. Pylance is starting, but it fails to find any user files. The problem is specific to UNC paths. It's not clear to me that we've ever supported UNC paths properly. A recent change that we made resulted in network drives (ex. `z:\Test`) being translated into their UNC equivalent (ex. `\\machine\share\Test`) and this is causing users to hit the UNC issue.

That seems to be exactly what is happening for me, thanks for the quick reply. Yep, it's trying to use the UNC path in 2023.9.20 and 2023.9.21, but uses the drive letter in 2023.9.10.

**Comment by Juliandev02** [comment]
I'm having the same issue with 2023.09.20. 
![image](https://github.com/microsoft/pylance-release/assets/81520713/a0e92bce-8752-48b0-864b-36dc48eb8185)
Just install another version.

**Comment by rchiodo** [comment]
This issue has been fixed in prerelease version 2023.9.31, which we've just released. You can find the changelog here: [CHANGELOG.md](https://github.com/microsoft/pylance-release/blob/main/CHANGELOG.md#2023931-27-September-2023-PreRelease)
